src/i965_avc_encoder_common.h

```diff
@@ -0,0 +1,321 @@
+ * OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT.
+ * IN NO EVENT SHALL PRECISION INSIGHT AND/OR ITS SUPPLIERS BE LIABLE FOR
+ * ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWAR OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * Authors:
+ *     Pengfei Qu <Pengfei.Qu@intel.com>
+ *
+ */
+
+#ifndef _I965_AVC_ENCODER_COMMON_H
+#define _I965_AVC_ENCODER_COMMON_H
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <math.h>
+#include <stdint.h>
+#include <assert.h>
+#include "intel_driver.h"
+#include "gen9_avc_encoder.h"
+
+// SubMbPartMask defined in CURBE for AVC ENC
+#define INTEL_AVC_DISABLE_4X4_SUB_MB_PARTITION    0x40
+#define INTEL_AVC_DISABLE_4X8_SUB_MB_PARTITION    0x20
+#define INTEL_AVC_DISABLE_8X4_SUB_MB_PARTITION    0x10
+#define INTEL_AVC_MAX_BWD_REF_NUM    2
+#define INTEL_AVC_MAX_FWD_REF_NUM    8
+
+#define MAX_MFC_AVC_REFERENCE_SURFACES      16
+#define NUM_MFC_AVC_DMV_BUFFERS             34
+#define MAX_HCP_REFERENCE_SURFACES      8
+#define NUM_HCP_CURRENT_COLLOCATED_MV_TEMPORAL_BUFFERS             9
+
+#define INTEL_AVC_IMAGE_STATE_CMD_SIZE    128
+#define INTEL_AVC_MIN_QP    1
+#define INTEL_AVC_MAX_QP    51
+
+#define INTEL_AVC_WP_MODE_DEFAULT 0
+#define INTEL_AVC_WP_MODE_EXPLICIT 1
+#define INTEL_AVC_WP_MODE_IMPLICIT 2
+
+struct avc_param {
+
+    // original width/height
+    uint32_t frame_width_in_pixel;
+    uint32_t frame_height_in_pixel;
+    uint32_t frame_width_in_mbs;
+    uint32_t frame_height_in_mbs;
+    uint32_t frames_per_100s;
+    uint32_t vbv_buffer_size_in_bit;
+    uint32_t target_bit_rate;
+};
+
+typedef enum
+{
+    INTEL_AVC_BASE_PROFILE               = 66,
+    INTEL_AVC_MAIN_PROFILE               = 77,
+    INTEL_AVC_EXTENDED_PROFILE           = 88,
+    INTEL_AVC_HIGH_PROFILE               = 100,
+    INTEL_AVC_HIGH10_PROFILE             = 110
+} INTEL_AVC_PROFILE_IDC;
+
+typedef enum
+{
+    INTEL_AVC_LEVEL_1                    = 10,
+    INTEL_AVC_LEVEL_11                   = 11,
+    INTEL_AVC_LEVEL_12                   = 12,
+    INTEL_AVC_LEVEL_13                   = 13,
+    INTEL_AVC_LEVEL_2                    = 20,
+    INTEL_AVC_LEVEL_21                   = 21,
+    INTEL_AVC_LEVEL_22                   = 22,
+    INTEL_AVC_LEVEL_3                    = 30,
+    INTEL_AVC_LEVEL_31                   = 31,
+    INTEL_AVC_LEVEL_32                   = 32,
+    INTEL_AVC_LEVEL_4                    = 40,
+    INTEL_AVC_LEVEL_41                   = 41,
+    INTEL_AVC_LEVEL_42                   = 42,
+    INTEL_AVC_LEVEL_5                    = 50,
+    INTEL_AVC_LEVEL_51                   = 51,
+    INTEL_AVC_LEVEL_52                   = 52
+} INTEL_AVC_LEVEL_IDC;
+
+/*
+common structure and define
+*/
+struct gen9_avc_encoder_context {
+
```

Comment:
I am thinking about supporting AVC on more platform, such as KBL/BDW etc and move this structure into the common. Yes, the structure name is gen9, i will change it to i965_avc_encoder_context to be common for other platforms.